### PR TITLE
Added authorize transaction support

### PR DESCRIPTION
**Pull request by galanis-a**
Added authorize transaction support.
Created config source for payment action.
Created payment action builder.
Removed payment action from general builder.

### CODE REVIEW

File: admin/Source/PaymentAction.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+declare(strict_types=1);
+
+namespace Pronko\LiqPayAdmin\Source;
+
+use Magento\Framework\Data\OptionSourceInterface;
+use Pronko\LiqPayApi\Api\Data\PaymentActionInterface;
+
+/**
+ * Class PaymentAction
+ * @package Pronko\LiqPayAdmin\Source
```
Comment (mcspronko) [inline admin/Source/PaymentAction.php]: Please remove @package comment.

File: admin/Source/PaymentAction.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ */
+class PaymentAction implements OptionSourceInterface
+{
+
+    /**
+     * Return array of options as value-label pairs
```
Comment (mcspronko) [inline admin/Source/PaymentAction.php]: Comment on line 17 can be removed.

File: admin/Source/PaymentAction.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+     * @return array Format: array(array('value' => '<value>', 'label' => '<label>'), ...)
+     */
+    public function toOptionArray()
+    {
+        return [
+            ['value' => PaymentActionInterface::AUTHORIZE, 'label' => __('Authorize Only')],
+            ['value' => PaymentActionInterface::AUTHORIZE_CAPTURE, 'label' => __('Authorize and Capture')],
+        ];
+    }
+}
```
Comment (mcspronko) [inline admin/Source/PaymentAction.php]: Please add empty line.

File: gateway/Gateway/Request/Builder/PaymentActionBuilder.php [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+namespace Pronko\LiqPayGateway\Gateway\Request\Builder;
+
+use Magento\Payment\Gateway\Request\BuilderInterface;
+use Pronko\LiqPayGateway\Gateway\Request\PaymentActionProvider;
+use Pronko\LiqPaySdk\Api\RequestFieldsInterface as RequestFields;
+
+/**
+ * Class PaymentActionBuilder
+ * @package Pronko\LiqPayGateway\Gateway\Request\Builder
```
Comment (mcspronko) [inline gateway/Gateway/Request/Builder/PaymentActionBuilder.php]: Please remove the @package comment.

File: gateway/Gateway/Request/Builder/PaymentActionBuilder.php [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+ */
+class PaymentActionBuilder implements BuilderInterface
+{
+
```
Comment (mcspronko) [inline gateway/Gateway/Request/Builder/PaymentActionBuilder.php]: Please remove an empty line.

File: gateway/Gateway/Request/Builder/PaymentActionBuilder.php [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+     * @param array $buildSubject
+     * @return array
+     */
+    public function build(array $buildSubject)
+    {
+        return [
+            RequestFields::ACTION => $this->actionProvider->getPaymentAction(),
+        ];
+    }
+}
```
Comment (mcspronko) [inline gateway/Gateway/Request/Builder/PaymentActionBuilder.php]: Please add the empty line.

**Comment by mcspronko** [comment]
@galanis-a all looks great, only a few formatting notes added. If you can have a look please.

**Comment by galanis-a** [comment]
Thanks Max,

done withe formatting. Please check.

File: card-gateway/etc/di.xml [hunk 7]
```diff
@@ -42,6 +42,7 @@
                 <item name="description" xsi:type="string">Pronko\LiqPayGateway\Gateway\Request\Builder\DescriptionBuilder</item>
                 <item name="order" xsi:type="string">Pronko\LiqPayGateway\Gateway\Request\Builder\OrderBuilder</item>
                 <item name="cardPayment" xsi:type="string">Pronko\LiqPayCardGateway\Gateway\Request\Builder\CardPaymentBuilder</item>
+                <item name="paymentAction" xsi:type="string">Pronko\LiqPayCardGateway\Gateway\Request\Builder\PaymentActionBuilder</item>
```
Comment (mcspronko) [inline card-gateway/etc/di.xml]: The class namespace is used "LiqPayCardGateway" however the class is located in the "LiqPayGateway" if you can fix this.

**Review by mcspronko — CHANGES_REQUESTED** [review]

**Comment by galanis-a** [comment]
Yeah sorry about that. Fixed

**Pull request merged**